### DISCUSSION TITLE
Add specific variables in order to set kubelet arguments: useful or not?

### DISCUSSION THREAD

**Discussion by cnd4** [kube-hetzner/terraform-hcloud-kube-hetzner#863] [Ideas]
While implementing a pull request to add custom `k3s agent` command arguments (see #862), I saw it would be easy to implement new variables (I implemented it on my side but I did not test it for now, I want to know if these variables are useful):

* adding a `kubelet_args` variable to set common kubelet arguments for control planes and agents nodes
* adding a `control_planes_kubelet_args` variable to set kubelet arguments for control planes nodes only 
* adding a `agents_kubelet_args` which variables to set kubelet arguments for agents nodes only

With the implementation I tried, these arguments are set directly in generated k3s configuration files (instead of passing arguments to `k3s server …` and `k3s agent …`.

My question is: would it be useful or does this idea conflict with `k3s_exec_server_args` (and the potential `k3s_exec_agent_args` variable I propose in #862)?

**Comment by mysticaltech** [comment]
@sambonbonne Because of the behavior layed out below. I think `k3s_exec_server_args` and `k3s_exec_agent_args` are enough. What do you think?

![ksnip_20230630-090851](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/assets/518555/17402ea8-44cd-44f9-86e2-9f836db7caea)

**Reply by cnd4** [reply]
@mysticaltech hey I fear this would be too repetitive. 

My implementation adds the `kubelet_args` in the `config.yaml` and the idea behind this idea is that it would be more "beginner friendly" to have simple variables for this specific purpose, but if think like this, we would have a specific variable for every setting in the k3s config file and it will be a lot of burden.

**Reply by mysticaltech** [reply]
Yes exactly, I mean, customizing k3s is not "beginner friendly" in any shape or form. So I think we are good as is and `k3s_exec_server_args` and `k3s_exec_agent_args` are enough, as you can do everything already.
